zuul-io-uring-test/src/test/java/com/netflix/zuul/netty/server/IoUringTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+package com.netflix.zuul.netty.server;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import com.netflix.netty.common.metrics.EventLoopGroupMetrics;
+import com.netflix.netty.common.status.ServerStatusManager;
+import com.netflix.spectator.api.NoopRegistry;
+import com.netflix.spectator.api.Spectator;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.group.DefaultChannelGroup;
+import io.netty.incubator.channel.uring.IOUring;
+import io.netty.incubator.channel.uring.IOUringSocketChannel;
+import io.netty.util.concurrent.GlobalEventExecutor;
+import static org.awaitility.Awaitility.await;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.netty.util.internal.PlatformDependent;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RunWith(JUnit4.class)
+public class IoUringTest {
+    private static final boolean IS_OS_LINUX = "linux".equals(PlatformDependent.normalizedOs());
+    private static final Logger LOGGER = LoggerFactory.getLogger(IoUringTest.class);
+
+    @Test
+    public void testIoUringServer() throws Exception {
```

Comment:
What's the goal of this test? Just verifying that the server starts up?

Reply:
I just added a JavaDoc comment that explains the goal of the test.